components/Person/EditablePersonPanel.js

```diff
@@ -27,10 +27,32 @@ export const EditablePersonPanel = ({ person }) => {
     setEditing(false)
   }
   const handleUpdate = async (person) => {
+    // Only a subset of fields can be updated on the server
+    const personData = {
```

Comment:
Yeah this is a tricky one, it all depends on the philosophy of how to secure fields on an API. I feel if a field is sent to the server which is not permitted to be updated then instead of the server ignoring that field and continuing, I instead return 403 forbidden. As the user has requested to update a field they're not allowed to, so they should be told about this.

In general it does make things a bit less forgiving for frontend developers because they need to be more strict with what they send. No longer can they just throw fields at the server and see what sticks.

An example I thought of where this more strict approach makes sense is; if we only want to update a single field, say `email`, and we `PUT` just that field as our body -
* If we ignore that field on the server (because the user is not allowed to update it), then we will still respond with 200 success because the server successfully updated 0 fields. This gives the impression to the frontend that their request worked and the `email` field was updated
* If we instead return forbidden because the `email` field cannot be updated, then this indicates more clearly that the request didn't do all that was asked for

So, for this specific line of code - when we GET a person from the server, it may contain fields which cannot be updated and are readonly (e.g. `dateAdded`). As these can no longer just be sent to the server, we either need to use a blacklist to trim readonly fields, or a whitelist to only send a specific set... I guess it's debatable which is preferred.